homeassistant/components/plugwise/climate.py

```diff
@@ -0,0 +1,250 @@
+        self._preset_modes = []
+        self._min_temp = min_temp
+        self._max_temp = max_temp
+        self._hvac_modes = ATTR_HVAC_MODES
+        self._hvac_mode = None
+        self._available_schemas = []
+
+        _LOGGER.debug("Initializing API")
+        import haanna
+
+        self._api = haanna.Haanna(
+            self._username, self._password, self._host, self._port
+        )
+        try:
+            self._api.ping_anna_thermostat()
+        except Exception:
+            _LOGGER.error("Unable to ping, platform not ready")
+            raise PlatformNotReady
+        _LOGGER.debug("Platform ready")
+        self.update()
+
+    @property
+    def should_poll(self):
+        """Return that we will require polling."""
+        return True
+
+    @property
+    def hvac_action(self):
+        """Return the current action."""
+        if self._api.get_heating_status(self._domain_objects) is True:
+            return CURRENT_HVAC_HEAT
+        return CURRENT_HVAC_IDLE
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        return DEFAULT_ICON
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        attributes = {}
+        attributes["outdoor_temperature"] = self._outdoor_temperature
+        attributes["available_schemas"] = self._api.get_schema_names(
+            self._domain_objects
+        )
+        attributes["active_schema"] = self._active_schema
+        attributes["previous_schema"] = self._previous_schema
+        return attributes
+
+    def update(self):
+        """Update the data from the thermostat."""
+        _LOGGER.debug("Update called")
+        self._domain_objects = self._api.get_domain_objects()
+        self._outdoor_temperature = self._api.get_outdoor_temperature(
+            self._domain_objects
+        )
+        api_active_schema = self._api.get_active_schema_name(self._domain_objects)
+        if self._active_schema != api_active_schema:
+            self._previous_schema = self._active_schema
+            self._active_schema = api_active_schema
+
+    @property
+    def hvac_mode(self):
+        """Return current active hvac state."""
+        if self._api.get_schema_state(self._domain_objects) is True:
+            return HVAC_MODE_AUTO
+        return HVAC_MODE_OFF
+
+    @property
+    def preset_mode(self):
+        """Return the active preset mode."""
+        return self._api.get_current_preset(self._domain_objects)
+
+    @property
+    def preset_modes(self):
+        """Return the available preset modes list without values."""
+        presets = list(self._api.get_presets(self._domain_objects).keys())
```

Comment:
Copying a dict to a list copies the keys by default. We can remove `.keys()`.

Reply:
Learning more and more, joy!